osu.Server.Queues.ScoreStatisticsProcessor/Processors/MedalAwarders/StarRatingMedalAwarder.cs

```diff
@@ -0,0 +1,154 @@
+﻿using JetBrains.Annotations;
+using MySqlConnector;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Difficulty;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Server.Queues.ScoreStatisticsProcessor.Helpers;
+using osu.Server.Queues.ScoreStatisticsProcessor.Models;
+using osu.Server.Queues.ScoreStatisticsProcessor.Stores;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Processors.MedalAwarders
+{
+    [UsedImplicitly]
+    [SuppressMessage("ReSharper", "MergeIntoPattern")]
+    public class StarRatingMedalAwarder : IMedalAwarder
+    {
+        private BeatmapStore? beatmapStore;
+
+        public bool RunOnFailedScores => false;
+
+        public IEnumerable<Medal> Check(SoloScoreInfo score, UserStats userStats, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            return checkAsync(score, medals, conn, transaction).Result;
+        }
+
+        private async Task<IEnumerable<Medal>> checkAsync(SoloScoreInfo score, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            List<Medal> earnedMedals = new List<Medal>();
+
+            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(score.RulesetID);
+            Mod[] mods = score.Mods.Select(m => m.ToMod(ruleset)).ToArray();
+
+            // Ensure the score isn't using any difficulty reducing mods
+            if (mods.Any(x => x is ModEasy || x is ModNoFail || x is ModHalfTime))
+                return earnedMedals;
```

Comment:
Spun Out too.

I'd recommend splitting this to a helper method because this is used very frequently in medal awarding code.